### DISCUSSION TITLE
UDMSE - Upgrade to 6.2.9 Linux Kernel

### DISCUSSION THREAD

**Discussion by CrazyTux** [fabianishere/udm-kernel-tools#91] [General]
Has anyone successfully upgraded the kernel UDM-SE ships with?

file /boot/firmware/rootfs:
/boot/firmware/rootfs: Squashfs filesystem, little endian, version 4.0, zstd compressed, 812082404 bytes, 98684 inodes, blocksize: 262144 bytes, created: Thu Feb 16 04:58:49 2023

I am planning on doing the following:

1. copy /boot/firmware/rootfs -> /volume1/rootfs
2. unsqaush /volume1/rootfs
3. chroot into copy of rootfs file system
4. compile 6.2.9 kernel (make olddefconfig)
5. replace kernel / make modules_install install
6. creating new squashfs "rootfs"
7. replacing /boot/firmware/rootfs with the new rootfs sqaushfs file system.

Any hard objections?  I've also looked at using unionfs, but would rather just do the above if there are no hard objections.  Any thoughts, considerations, or input anyone can add in doing this?

Thanks everyone in advance for your time.

**Comment by fabianishere** [comment]
I've investigated this and the problem is that the device requires multiple drivers which are not included in the mainline kernel (e.g., PCIe, Ethernet, AHCI, IOFIC). You might be able to get the device to boot, but the network interfaces and probably storage won't work.

You can find the drivers for the stock kernel (4.19.152) here: https://github.com/fabianishere/udm-kernel (look for drivers with alpine or al in the name).
